src/wallet/wallet.cpp

```diff
@@ -2815,9 +2820,9 @@ bool CWallet::CreateTransactionInternal(
 
                 // Never create dust outputs; if we would, just
                 // add the dust to the fee.
-                // When nChange is less than the cost of the change output,
-                // send it to fees (this means BnB was used)
-                if (IsDust(newTxOut, discard_rate) || nChange <= cost_of_change)
+                // When the selected_value is within the exact match range
```

Comment:
Not immediately obvious to me what this change is doing, explain? Also this should say `selected_eff`?

Reply:
This is essentially doing what `used_bnb` did.

If the selected effective value is within the exact matching range that we use for BnB, then let the change go to fees.